### PR TITLE
audit & fix test_timeline_delete.py's usage of ps_http.timeline_delete()

### PR DESCRIPTION
**Pull request by problame**
(For https://github.com/neondatabase/neon/pull/4495)

Most of the `test_delete_timeline.py` tests make negative tests, i.e., "does `ps_http.timeline_delete()` fail in this and that scenario".
These can be left alone.

But when we actually do the deletions, we need to use the helper that polls for completion.

### CODE REVIEW

**Review by LizardWizzard — APPROVED** [review]
Thanks! Makes sense. Initially I didnt change those, but after re-reading it makes sense.

See one comment

File: test_runner/regress/test_timeline_delete.py [hunk 1]
```diff
@@ -96,13 +96,6 @@ def test_timeline_delete(neon_simple_env: NeonEnv):
     ) as exc:
         ps_http.timeline_detail(env.initial_tenant, leaf_timeline_id)
 
-        # FIXME leaves tenant without timelines, should we prevent deletion of root timeline?
-        wait_until(
-            number_of_iterations=3,
-            interval=0.2,
-            func=lambda: ps_http.timeline_delete(env.initial_tenant, parent_timeline_id),
-        )
```
[the remark this answers is not in the document]
Comment (problame) [reply test_runner/regress/test_timeline_delete.py:104]: It doesn't make sense, the timeline_detail above is the one that should be 404'ing.
We delete&wait'ed the timeline in the block above.

[the remark this answers is not in the document]
Comment (LizardWizzard) [reply test_runner/regress/test_timeline_delete.py:104]: Note that detail uses `leaf_timeline_id` and this delete uses `parent_timeline_id`. Should we move it out of  the `raises` context manager then?

[the remark this answers is not in the document]
Comment (problame) [reply test_runner/regress/test_timeline_delete.py:104]: > Note that detail uses leaf_timeline_id and this delete uses parent_timeline_id.

oops, I missed that

> Should we move it out of the raises context manager then?

Yeah, I'll push a fix here.

File: test_runner/regress/test_timeline_delete.py [hunk 2]
```diff
@@ -95,16 +95,14 @@ def test_timeline_delete(neon_simple_env: NeonEnv):
...
-
     assert exc.value.status_code == 404
 
+    wait_until(
+        number_of_iterations=3,
+        interval=0.2,
+        func=lambda: ps_http.timeline_delete(env.initial_tenant, parent_timeline_id),
```
Comment (LizardWizzard) [inline test_runner/regress/test_timeline_delete.py:103]: I think should be also `timeline_delete_wait_completed` (the same as above)

**Review by LizardWizzard — APPROVED** [review]

**Pull request merged**